lib/resque/worker.rb

```diff
@@ -362,7 +362,7 @@ def prune_dead_workers
     # lifecycle on startup.
     def register_worker
       redis.sadd(:workers, self)
-      started!
+      redis.set("worker:#{self}:queues", @queues.join(","))
```

Comment:
Also, totally fine with a separate PR and only if necessary. Separate PR might even be better to see the difference this one makes on its own and then see the diff on the pipelining.